### DISCUSSION TITLE
Sobald ich MQTT einschalte verbindet sich der ESP32 nicht mehr mit meinem Wlan / nur noch AP Setup Mode

### DISCUSSION THREAD

**Discussion by Knicko83** [dewenni/ESP_Buderus_KM271#79] [Q&A] [answered]
Hi in die Nachbarschaft,
habe deine Software mit einer originalen KM271 schon seit Längerem am Laufen (genial!) mit 3.0.1 und MQTT zu meinem IOBroker und heute Nachmittag festgestellt, dass ich einige Versionen verpasst habe. Also mit Zwischenschritt 3.2.5 zum Sichern der Config auf neuestes Release 3.3.1 geupdated - und jetzt leider den gesamten Abend im Heizungskeller verbracht, weil ich das Board einfach nicht mehr zum Laufen bringe.

Folgendes Symptom: ich flashe, ESP startet mit km271-Accesspoint, ich logge mich ein, gebe mein Wlan SSID/passwort an. Board bootet neu, ist im Wlan und per Browser ansprechbar. Ich kann die Heizungswerte sehen, Settings ändern, alles gut. Sobald ich aber die MQTT-Verbindung aktiviere und abspeichere, ist Schluss. Ich kann im IObroker-Log sehen, dass sich das Board einmalig versucht mit dem MQTT-Broker zu verbinden, bevor die Verbindung direkt wieder abgebrochen und auch nicht wieder aufgebaut wird.

Das Board bootet neu und bleibt permanent mit offenem km271-Accesspoint hängen, und verbindet sich nicht mehr mit dem Wlan. Es ist über 192.168.4.1 ansprechbar, aber es verbindet sich nicht mehr mit dem Wlan. Die Konfig-Daten habe ich wohlgemerkt nicht geändert. Ich hab das mehrere Male ausprobiert, mit 3.3.1, 3.2.5, 3.2.1 - immer genau dasselbe Verhalten: sobald ich MQTT aktiviere, schießt sich das Board für ewig aus dem Wlan und geht beim Booten direkt in den km271-Setup-Accesspoint über. Kein Verbindungsversuch mehr in meiner Fritzbox sichtbar. Die Wlan-Zugangsdaten sind korrekt, und auch erneutes eingeben oder Neustart ändert nichts an der Situation.

Habe zwischendurch den Flash komplett gelöscht, mit leerem Image überschrieben, direkt per COM geflasht über Espressif-Tool, per OTA, und auch selbst kompiliert in Platformio und dann per COM hochgeladen - Verhalten haargenau identisch und reproduzierbar.

Ich hab jetzt verzweifelt nach Stunden wieder auf 3.0.1 downgegradet, und dasselbe Board funktioniert wieder problemlos - mit haargenau den selben Settings für Wlan und MQTT und sonstiger Config.

Ich bin ehrlich gesagt ratlos, und hab keine Idee wo und wie ich mit der weiteren Fehlersuche anfangen kann. Any idea, woran das liegen könnte?

**Comment by dewenni** [comment]
Bezüglich mqtt gab es aber der [v3.2.1](https://github.com/dewenni/ESP_Buderus_KM271/releases/tag/v3.2.1) eine Änderung. 
Vielleicht bringt diese Änderung auch ein geändertes Verhalten in Zusammenhang mit dem ioBroker mit sich?!

Kommt es auch dann zum Fehler wenn du andere mqtt Parameter verwendest, die nicht im ioBroker konfiguriert sind?
Mich würde interessieren ob das nicht sogar mit dem ioroker zusammenhängt.

Wir hatten hier schon mal ein vergleichbares verhalten #36 

schau auch mal hier #38 und #58

**Comment by joeflower** [comment]
Ich nutze auch den MQTT Adapter im IObroker. Funktioniert in der 3.3.1 (und vorher auch) tadellos.
![grafik](https://github.com/dewenni/ESP_Buderus_KM271/assets/17798165/5524ad1d-54d4-4fb2-a6cb-9998a2207586)

**Comment by Knicko83** [comment]
Hi, danke für die schnelle Antwort.
Wenn ich das ioBroker-Log aus #36 so anschaue, ist das bei mir quasi genau gleich. Auch habe ich die Option
![grafik](https://github.com/dewenni/ESP_Buderus_KM271/assets/30951261/80345eb5-39bc-49e3-92aa-5d6a708f2a1c)
bei mir im MQTT-Adapter aktiviert, weil einige meiner anderen Anwendungen das aus Prozessgründen brauchen, dass Settings gleich bei Verbindungsaufbau mitgeschickt werden, und nicht nur wenn diese sich wieder ändern.

Ich werde das mal mit einem zweiten ESP-Board und einem eigenständigen MQTT-Adapter testen bei nächster Gelegenheit.
Memo an mich: testen
-ob Unterschied, wenn "MQTT Publish on subsribe" ausgschaltet
-ob Unterschied, wenn Kommandorichtung MQTT iOB->KM271 deaktiviert
-ob Unterschied, wenn neues Topic beim Senden

Ich geb Rückmeldung, wenn ich schlauer bin!

**Comment by Knicko83** [comment]
So, gerade rumprobiert.

-Frisches Image 3.3.1 geflashed (kein KM271 angeschlossen)
-Wlan Verbindung OK
-MQTT aktiviert - nur Schalter umgelegt, keine Daten eingetragen (Default Settings) -> keine Verbindung ins WLAN, permanenter KM271-AP
-MQTT wieder deaktiviert - Verbindet sich wieder mit WLAN
-MQTT settings erfundene IP (nichtexistent) mit erfundener Config eingetragen - verbindet sich mit WLAN
-KM271-Web-IF- Settings alle auf Endzustand - funktioniert immer noch
-Dedizierte MQTT-Test-Instanz in IOB erstellt - Standardeinstellungen - Verbindung funktioniert

So, und jetzt hab ich viel rumprobiert. Korrektur von gestern: der ESP verbindet sich immer ganz kurz mit dem Wlan, und per MQTT mit dem IOB, aber bricht danach die Verbindung ab und macht den Accesspoint auf.
![grafik](https://github.com/dewenni/ESP_Buderus_KM271/assets/30951261/ed7d15b9-68b6-45e3-9cf5-34b66e03c592)

Ich glaube, ich konnte mein Problem leicht eingrenzen, so richtig schlau werde ich aber nicht draus.
Die Software scheint sich beim abonnieren von den setvalue-Variablen zu verschlucken:
Ich habe versucht, Kommandovariablen zu erstellen zum steuern; genau wie dich das mit der alten KM271-Version mache.
Danach immer das Board hard-resettet und geschaut, ob's hinterher noch funktioniert.

![grafik](https://github.com/dewenni/ESP_Buderus_KM271/assets/30951261/0c033265-216a-4652-92fe-9d0c8af515df)
-Variable hk1_betriebsart mit Wert 0 angelegt: alles noch gut, Verbindung WLAN/MQTT stabil
-Variable hk1_betriebsart auf 1 gesetzt: KM271 Accesspoint - Fehler
Ich hab ein paarmal hin und her probiert; wenn ich die Variable lösche und ein paarmal neu starte, dann verbindet er sich undefiniert wieder. Wenn ich sie dann neu anlege, gefühlt ähnliches Verhalten. Sobald ich sie auf 1 setze, ist er immer wieder weg. So richtig verstehe ich aber nicht, was ich tun muss, damit es irgendwann wieder funktioniert hat.

Eventuall hab ich irgendwas falsch konfiguriert und seh's nicht? Der Haken in den IOB-Einstellungen bei "States beim Verbinden publizieren" hat bei mir gefühlt keinen Unterschied gemacht.

Habe unten einen Serial Dump angehängt vom Zeitpunkt, wo er den Fehler wirft und in den AP Mode geht - eventuell hilft das ja weiter. 

`LittleFS Status: 1
WiFi Mode STA - Trying connect to: FRITZ!Box7590
UI Initialized
Webserver started
Widgets: 241
Connected to AP successfully!
WiFi connected
IP address:
192.168.179.31
MQTT - connection attempt: 1/5
MQTT connected
topic: test/setvalue/ww_soll
MQTT disconnected
Guru Meditation Error: Core  1 panic'ed (LoadProhibited). Exception was unhandle                                                              d.

Core  1 register dump:
PC      : 0x400f455a  PS      : 0x00060330  A0      : 0x800f45c0  A1      : 0x3f                                                              fe9410
A2      : 0x3ffc73b4  A3      : 0x00000000  A4      : 0x3ffc74b4  A5      : 0x00                                                              000003
A6      : 0x3ffebafe  A7      : 0x0000003d  A8      : 0x00000000  A9      : 0x00                                                              000074
A10     : 0x00000000  A11     : 0x00000001  A12     : 0x00000003  A13     : 0x40                                                              181bf0
A14     : 0x3ffead14  A15     : 0x00000000  SAR     : 0x00000009  EXCCAUSE: 0x00                                                              00001c
EXCVADDR: 0x00000000  LBEG    : 0x4008a025  LEND    : 0x4008a035  LCOUNT  : 0xff                                                              fffffb


Backtrace: 0x400f4557:0x3ffe9410 0x400f45bd:0x3ffe9480 0x4016d359:0x3ffe94a0 0x4                                                              016d3d1:0x3ffe94d0 0x4016db82:0x3ffe94f0




ELF file SHA256: 082637654c567dc2

Rebooting...
ets Jul 29 2019 12:21:46

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:1344
load:0x40078000,len:13836
load:0x40080400,len:3608
entry 0x400805f0
LittleFS Flag read = 0xD0D01234
                               doubleResetDetected
Saving config file...
Saving config file OK
LittleFS Status: 1

! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! !

> WiFi Mode: AccessPoint <
1. connect your device to SSID: ESP-Buderus-km271
2. open Browser and go to Address: http://192.168.4.1

! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! !

UI Initialized
Webserver started
Widgets: 79
`

**Comment by Kenaschon** [comment]
Bin mir nicht sicher ob das für die Problemlösung behilflich ist,
aber ich habe auch ioBroker hier auf einem Raspberry 4 laufen. Der ESP-Buderus ist per MQTT mit ioBroker verbunden.

Ich kann folgendes Verhalten reproduzieren.

Wenn ich den MQTT-Adapter unter ioBroker deaktiviere (auf Pause schalte) verliert der ESP-Buderus nach kurzer Zeit die Wlan Verbindung.

Das kann ich immer wieder machen. Ist der Adapter wieder an - läuft der ESP ohne Probleme.

Grüße
Manfred

**Comment by dewenni** [comment]
Danke für deine Tests.
Der Log hat mir einen entscheidenden Hinweis gegeben.
Es gibt ein Problem wenn eine MQTT Nachricht ohne payload empfangen wird.

Das habe ich in dem 3.4.0rc #74
behoben. Kannst du die bitte mal testen.
Dann mache ich gerne auch noch einen schnellen fix für die 3.3.1

**Reply by dewenni** [reply]
wenn du das Projekt in platformio am laufen hast, kannst du testweise auch nur den fix aus dem entsprechenden Commit einbauen
=> https://github.com/dewenni/ESP_Buderus_KM271/commit/cf4b2a7e99269f082f56a775271e04336e4238b0

**Reply by Knicko83** [reply]
Sorry, war beruflich bis gerade busy und konnte nicht nachverfolgen. Werd ich heute Abend testen und geb Rückmeldung! Ich drück die Daumen

**Reply by Knicko83** [reply, marked as answer]
Hey @dewenni, also von meiner Seite aus Daumen hoch! 3.4.0rc geflashed und stabile Verbindung in MQTT und WLAN - krieg's nicht mehr tot :) - beobachtetes Verhalten bei mir weg.
Top! *kudos*

**Comment by Kenaschon** [comment]
So, ich habe jetzt die 3.4.0rc aufgespielt.

Der Fehler ist immer noch da. Was dazu gekommen ist, ist das die Wlan-Verbindung von ganz alleine nach sehr kurzer Zeit abbricht. Das hatte ich mit der 3.1.1 nicht.

Grüße
Manfred

**Reply by dewenni** [reply]
du sprichst aber von dem Problem:
_Wenn ich den MQTT-Adapter unter ioBroker deaktiviere (auf Pause schalte) verliert der ESP-Buderus nach kurzer Zeit die Wlan Verbindung._

Das sehe ich nochmal getrennt von dem Problem, dass @Knicko83  berichtet.

**Comment by Kenaschon** [comment]
Jepp,

Besser einen neuen Beitrag aufmachen?
